### PR TITLE
strict options in tests; few minor enhancements

### PR DESCRIPTION
**Pull request by pawelru**
- part of https://github.com/insightsengineering/coredev-tasks/issues/478
  - please read this for more info about the implementation: https://github.com/insightsengineering/coredev-tasks/issues/478#issuecomment-1909912778
- added more excludes in spelling hook
- removed unnecessary empty first line in examples as this is actually being rendered
- reordered roxy tags to have examples at the bottom
- reviewed `dontrun{}` clauses
- removed `formatters::` prefix in tests as this is redundant inside `formatters`

Please review the changes carefully and let me know if there is something you don't like.

### CODE REVIEW

File: R/page_size.R [hunk 1]
```diff
@@ -87,16 +87,14 @@ page_dim <- function(page_type) {
 #' @return named list with `cpi` and `lpi`, the characters and lines per
 #' inch, respectively.
 #'
+#' @keywords internal
```
Comment (Melkiades) [inline R/page_size.R:90]: also this I think it is better to have close to the function declaration

Comment (pawelru) [reply R/page_size.R]: ok moved

File: R/tostring.R [hunk 2]
```diff
@@ -1120,20 +1119,19 @@ propose_column_widths <- function(x, indent_size = 2) {
...
 #' padstr("abc", 3)
 #' padstr("abc", 4)
 #' padstr("abc", 5)
 #' padstr("abc", 5, "left")
 #' padstr("abc", 5, "right")
 #'
-#' if (interactive()) {
+#' \dontrun{
 #'   padstr("abc", 1)
 #' }
```
Comment (shajoezhu) [inline R/tostring.R:1134]:
```suggestion
#' \dontrun{
#' # Expect error: "abc" has more than 1 characters
#'   padstr("abc", 1)
#' }
```

**Review by shajoezhu — APPROVED** [review]
lgtm! thanks @pawelru

**Pull request merged**